setup.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# Update the dependencies
+sudo apt-get update
+
+# Check if Docker is installed
+if ! command -v docker &> /dev/null; then
+    echo "Docker is not installed. Installing Docker..."
+    
+    # Install Docker dependencies
+    sudo apt-get install -y apt-transport-https ca-certificates curl software-properties-common
+    
+    # Add Docker's official GPG key
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+    
+    # Add Docker repository
+    echo "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+    
+    # Update the package index (again)
+    sudo apt-get update
+    
+    # Install Docker
+    sudo apt-get install -y docker-ce docker-ce-cli containerd.io
+
+    # Add current user to docker group
+    sudo groupadd docker
+    sudo usermod -aG docker $USER
+    
+    echo "Docker installed successfully!"
+else
+    echo "Docker is already installed."
+fi
+
+sudo apt-get install -y python3 python3-pip
+
+# Install base dependencies for Ansible roles
+sudo apt install ansible
```

Comment:
How are you planning to call ansible commands from the jenkins container ?

Reply:
I think we need to make a custom Jenkins container with Ansible installed.